TypeCobol/Analysis/AnalyzerProviderLoader.cs

```diff
@@ -17,19 +19,28 @@ public static class AnalyzerProviderLoader
         /// </summary>
         /// <param name="compositeAnalyzerProvider">CompositeAnalyzerProvider instance into which the external analyzers should be added.</param>
         /// <param name="assemblyFilePaths">List of paths of .NET Assembly files.</param>
-        public static void AddCustomProviders(this CompositeAnalyzerProvider compositeAnalyzerProvider, IEnumerable<string> assemblyFilePaths)
+        /// <param name="logFilePath">Path of a non-null log file.</param>
+        public static void AddCustomProviders(this CompositeAnalyzerProvider compositeAnalyzerProvider, IEnumerable<string> assemblyFilePaths, [NotNull] string logFilePath)
         {
             if (compositeAnalyzerProvider == null || assemblyFilePaths == null) return;
 
             foreach (var assemblyFilePath in assemblyFilePaths)
             {
-                var provider = LoadProvider(assemblyFilePath);
-                compositeAnalyzerProvider.AddProvider(provider);
+                try
+                {
+                    var provider = LoadProvider(assemblyFilePath);
+                    compositeAnalyzerProvider.AddProvider(provider);
+                }
+                catch (Exception exception)
+                {
+                    File.AppendAllText(logFilePath, $"Failed to load analyzer provider on path {assemblyFilePath}\n{exception.Message}\n{exception}");
```

Comment:
Use `Environment.NewLine` instead of `\n` or use `File.AppendAllLines` to let the framework determine the newline separator. As for the error message, I would rather have

"Failed to load analyzer provider from path [path]."
"[exception.GetType().FullName] has been thrown."
"[exception.Message]"

Note: using `exception` in interpolated string will call its `ToString` method which usually return the exception type name but it may be redefined.

Reply:
Message has been updated as suggested.